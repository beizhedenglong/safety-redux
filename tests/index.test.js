import { render, fireEvent } from 'react-testing-library'
import React from 'react'
import App from '../example/App'


test('safety-redux', () => {
  const { container, queryByText } = render(<App />)
  const addButton = queryByText(/\+1/)
  const subtractButton = queryByText(/-1/)
  expect(container).toMatchSnapshot()
  expect(queryByText(/Age:/).textContent).toContain('20')
  fireEvent.click(addButton)
  expect(queryByText(/Age:/).textContent).toContain('21')
  fireEvent.click(subtractButton)
  expect(queryByText(/Age:/).textContent).toContain('20')
  fireEvent.click(subtractButton)
  expect(queryByText(/Age:/).textContent).toContain('19')
})
