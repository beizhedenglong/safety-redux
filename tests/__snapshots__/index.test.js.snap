// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`safety-redux 1`] = `
<div>
  <div>
    <span>
      Name: Victor
    </span>
    <br />
    <span>
      Age: 20 
    </span>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
  </div>
</div>
`;
